docs/Compile.rst

```diff
@@ -200,39 +280,141 @@ On Windows, DFHack replaces the SDL library distributed with DF.
 
 Dependencies
 ------------
-You will need some sort of Windows port of git, or a GUI. Some examples:
+You will need the following:
+
+* Microsoft Visual Studio 2010 SP1, with the C++ language
+* Git
+* CMake
+* Perl with XML::LibXML and XML::LibXSLT
+
+  * It is recommended to install StrawberryPerl, which includes both.
+
+Microsoft Visual Studio 2010 SP1
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+The free Express version is sufficient.
+
+You can grab it from `Microsoft's site <http://download.microsoft.com/download/1/E/5/1E5F1C0A-0D5B-426A-A603-1798B951DDAE/VS2010Express1.iso>`_.
+
+You'll also need the Visual Studio 2010 SP1 update, which is obtained from
+Windows Update. After installing Visual Studio, be sure to go to Windows Update
+and check for and install the SP1 update. If no update is found, check that
+your Windows Update settings include "Updates from all Microsoft products".
+
+You can confirm whether you have SP1 by opening the Visual Studio 2010 IDE
+and selecting About from the Help menu.  If you have SP1 it will have *SP1Rel*
+at the end of the version number, for example: *Version 10.0.40219.1 SP1Rel*
+
+It is vital that you use SP1 as, while building with the original release
+of Visual Studio 2010 (RTM) may succeed, it will result in non-working DFHack
+binaries that crash when connecting to Dwarf Fortress.
```

Comment:
Ah, I thought it was ABI issues on the basis that it *does* compile on RTM (non-SP1).   At least someone in #dfhack got it compiled the other day on RTM.  He did have a bunch of issues but ultimately a lot of that may have been his own system issues.

If you've heard it's not ABI issues, and I've heard it's not compilation issues, then maybe it's no issue at all.

I can try it out I guess, but it's so easy for anyone to upgrade to SP1 I'll probably just tone down the above message to something like "Using SP1 may cause issues, don't do it." rather than stating what they are.